rel/pattern.go

```diff
@@ -165,3 +160,66 @@ func (p TuplePattern) String() string {
 	b.WriteByte(')')
 	return b.String()
 }
+
+type DictPatternEntry struct {
```

Comment:
I was going to suggest that this is really a `DictEntryPattern`. But actually, it's both. :grin: